### PR TITLE
enh: store basic software info in png metadata

### PR DESCRIPTION
**Pull request by Xarthisius**
## PR Summary

Store information about used software (yt, matplotlib, numpy, PIL) in metadata of generated plots.

## PR Checklist

- [ ] New features are documented, with docstrings and narrative docs
- [x] Adds a test for any bugs fixed. Adds tests for new features.

## How to test?
1. Generate simple SlicePlot, ProjectionPlot, off-axis projection / VR png.
2. Use e.g. ImageMagick to inspect resuting pngs:
```
$ identify -verbose galaxy0030_offaxis_projection.png | grep Software
    Software: PIL-9.5.0|yt-4.3.dev0
$ identify -verbose *.png | grep Software
    Software: Matplotlib version3.7.1, https://matplotlib.org|NumPy-1.24.2|yt-4.3.dev0
```

### CODE REVIEW

**Comment by Xarthisius** [comment]
pre-commit.ci autofix

**Comment by neutrinoceros** [comment]
love the idea !

**Review by matthewturk — DISMISSED** [review]
strong approve

**Comment by Xarthisius** [comment]
pre-commit.ci autofix

**Review by neutrinoceros — COMMENTED** [review]
Maybe `InteractiveCamera.save` could use some of this too, but I couldn't figure out how, and I'm not even sure it's actually feasible so it's not crucial. Other than that, I only have a couple minor comments

**Comment by chrishavlin** [comment]
Cool! This is great! Don't have anything to add to the review... but I was playing around locally and devised a little self contained python-only example, might be useful for adding a test:

```python
from yt.testing import fake_amr_ds
from yt import SlicePlot
import PIL

ds = fake_amr_ds()
fn = SlicePlot(ds, "x", "Density").save()[0]

with PIL.Image.open(fn) as image:
    assert 'Software' in image.text
    assert 'Matplotlib version' in image.text['Software']
```

File: yt/funcs.py [hunk 1]
```diff
@@ -1447,3 +1448,25 @@ def validate_moment(moment, weight_field):
             "Weighted projections can only be made of averages "
             "(moment = 1) or standard deviations (moment = 2)!"
         )
+
+
+def setdefault_mpl_metadata(mpl_kwargs: Dict[str, Any], name: str) -> None:
```
[the remark this answers is not in the document]
Comment (Xarthisius) [reply yt/funcs.py:1453]: It's specifically a matplotlib kwargs `{"metadata": {...}}`, not just yt's save.

File: yt/visualization/tests/test_save.py [hunk 2]
```diff
@@ -23,6 +24,16 @@ def test_save_to_path(simple_sliceplot, tmp_path):
     assert len(list((tmp_path).glob("*.png"))) == 1
 
 
+def test_metadata(simple_sliceplot, tmp_path):
+    simple_sliceplot.save(f"{tmp_path}/ala.png")
+    with Image.open(f"{tmp_path}/ala.png") as img:
```
Comment (neutrinoceros) [inline yt/visualization/tests/test_save.py:29]:
```suggestion
    with Image.open(tmp_path / "ala.png") as img:
```

File: yt/visualization/tests/test_save.py [hunk 3]
```diff
@@ -23,6 +24,16 @@ def test_save_to_path(simple_sliceplot, tmp_path):
...
+        assert "Software" in img.info
+        assert "yt-" in img.info["Software"]
+    simple_sliceplot.save(f"{tmp_path}/ala.pdf")
+    with open(f"{tmp_path}/ala.pdf", "rb") as f:
```
Comment (neutrinoceros) [inline yt/visualization/tests/test_save.py:33]:
```suggestion
    with open(tmp_path / "ala.pdf", "rb") as f:
```

File: yt/visualization/tests/test_save.py [hunk 4]
```diff
@@ -23,6 +24,16 @@ def test_save_to_path(simple_sliceplot, tmp_path):
     assert len(list((tmp_path).glob("*.png"))) == 1
 
 
+def test_metadata(simple_sliceplot, tmp_path):
+    simple_sliceplot.save(f"{tmp_path}/ala.png")
```
Comment (neutrinoceros) [inline yt/visualization/tests/test_save.py:28]:
```suggestion
    simple_sliceplot.save(tmp_path / "ala.png")
```

**Review by neutrinoceros — APPROVED** [review]

**Pull request merged**